src/Digdir.Domain.Dialogporten.Application/Features/V1/ResourceRegistry/Commands/Synchronize/SynchronizeResourceRegistryCommand.cs

```diff
@@ -0,0 +1,142 @@
+{
+    private readonly IDialogDbContext _db;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IResourceRegistry _resourceRegistry;
+    private readonly ILogger<SynchronizeResourceRegistryCommandHandler> _logger;
+
+    public SynchronizeResourceRegistryCommandHandler(
+        IDialogDbContext db,
+        IUnitOfWork unitOfWork,
+        IResourceRegistry resourceRegistry,
+        ILogger<SynchronizeResourceRegistryCommandHandler> logger)
+    {
+        _db = db ?? throw new ArgumentNullException(nameof(db));
+        _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+        _resourceRegistry = resourceRegistry ?? throw new ArgumentNullException(nameof(resourceRegistry));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+    public async Task<SynchronizeResourceRegistryResult> Handle(SynchronizeResourceRegistryCommand request, CancellationToken cancellationToken)
+    {
+        // 1. Get the last updated timestamp from parameter, or the database, or use a default
+        var lastUpdated = request.Since
+            ?? (await _db.SubjectResourceLastUpdates
+                .OrderByDescending(x => x.LastUpdate)
+                .FirstOrDefaultAsync(cancellationToken))
+                ?.LastUpdate
+            ?? DateTimeOffset.MinValue;
+
+        // Allow for a few seconds of drift between server clocks
+        var newLastUpdated = DateTimeOffset.Now.Subtract(TimeSpan.FromSeconds(3));
+
+        _logger.LogInformation("Fetching updated subject resources since {LastUpdated}", lastUpdated);
+
+        // 2. Get the resources from the resource registry, supplying the last updated as "since"
+        var remoteUpdatedSubjectResources =
+            await _resourceRegistry.GetUpdatedSubjectResources(lastUpdated, cancellationToken);
+
+        _logger.LogInformation("Fetched {Count} updated subject resources", remoteUpdatedSubjectResources.Count);
+
+        // 3. Delete the subjectResources that are no longer in the resource registry
+        var remoteDeletedSubjectResources = remoteUpdatedSubjectResources
+            .Where(x => x.Deleted)
+            .Select(deletedSubjectResource => new
+            {
+                deletedSubjectResource.Resource,
+                deletedSubjectResource.Subject
+            })
+            .ToList();
+
+        if (remoteDeletedSubjectResources.Count != 0)
+        {
+            var resourceStrings = remoteDeletedSubjectResources.Select(r => r.Resource.ToString()).ToList();
+            var subjectStrings = remoteDeletedSubjectResources.Select(r => r.Subject.ToString()).ToList();
+
+            var localSubjectResourcesToDelete = await _db.SubjectResources
+                .Where(x => resourceStrings.Contains(x.Resource) && subjectStrings.Contains(x.Subject))
+                .ToListAsync(cancellationToken);
```

Comment:
Et par ting her. 
1. Er det ikke kombinasonen av ressurs og subjekt vi vil slette? Si at vi har følgende ressurs- og subjekt-grupper i DB: (r1, s1), (r1, s2), (r1, s2), (r2, s2).  Følgende er markert for sletting: (r1, s1) og (r2, s2). Denne koden vil fjerne de riktige gruppene, i tillegg til (r1, s2), og (r2, s1). Er dette korrekt? 
2. Kan det komme flere UpdateSubjectResource som representerer den samme gruppa, og blir det hensyntatt? Si (r1, s1) er markert slettet kl 12:00, (gjen)opprettet kl 12:10, også slettet igjen kl 12:20. Da vil denne delen av koden først fjerne (r1, s1), også opprette den igjen lenger ned (sant?). Men den skal egentlig bare være slettet.
3. Jeg tror at vi i dette tilfellet kan jobbe direkte i DB gjennom ExecuteDeleteAsync i stede for å laste det inn i minnet og slette det gjennom UnitOfWork. Egentlig tror jeg vi kan skippe alle Entity rammene vi har for disse tabellene.

Reply:
1. Skal se på denne.
2. Et gitt par vil aldri kunne dukke opp mer enn én gang i et changeset. Tabellen dette hentes fra er ikke en logg, men en tabell med constraints på at hvert par er unike med timestamps og softdeletes.
3. Tror også hele denne rutinen hadde blir vesenlig enklere (og raskere) med ren SQL. Skal se på det.

Reply:
Droppet EF helt her til fordel for direkte bruk av Npgsql, som muliggjør bruk av arrays av tuples/unnest for å unngå looping. La inn property til `DatabaseFacade` i `IDialogDbContext`, slik at vi kan nå connectionstringen for å lage en egen connection.